### PR TITLE
Module Registry removeModule() function

### PR DESCRIPTION
**Pull request by thegostep**
Please double check logic for dealing with array index 0. My approach has been to set index 0 of the array to the 0x0 address and prevent from removing, the side effect is that getting the list of module factories will always include the 0x0 address.

### CODE REVIEW

**Comment by pabloruiz55** [comment]
@thegostep why not just skip the whole re-ordering code block if array length is == 1?

File: contracts/ModuleRegistry.sol [hunk 1]
```diff
@@ -20,6 +20,8 @@ contract ModuleRegistry is IModuleRegistry, Pausable, RegistryUpdater, ReclaimTo
     mapping (address => address[]) public reputation;
     // Mapping contain the list of addresses of Module factory for a particular type
     mapping (uint8 => address[]) public moduleList;
+    // Mapping to store the index of the moduleFacorty in the moduleList
```
Comment (pabloruiz55) [inline contracts/ModuleRegistry.sol]: typo => moduleFactory

File: contracts/ModuleRegistry.sol [hunk 2]
```diff
@@ -67,14 +75,47 @@ contract ModuleRegistry is IModuleRegistry, Pausable, RegistryUpdater, ReclaimTo
     function registerModule(address _moduleFactory) external whenNotPaused returns(bool) {
         require(registry[_moduleFactory] == 0, "Module factory should not be pre-registered");
         IModuleFactory moduleFactory = IModuleFactory(_moduleFactory);
-        require(moduleFactory.getType() != 0, "Factory type should not equal to 0");
-        registry[_moduleFactory] = moduleFactory.getType();
-        moduleList[moduleFactory.getType()].push(_moduleFactory);
+        uint8 kind = moduleFactory.getType();
```
Comment (pabloruiz55) [inline contracts/ModuleRegistry.sol]: instead of "kind" use "type"

Comment (thegostep) [reply contracts/ModuleRegistry.sol]: type is a reserved keyword [documentation](https://solidity.readthedocs.io/en/v0.4.21/miscellaneous.html#reserved-keywords)

**Comment by thegostep** [comment]
~~the check on line 99 does the same implicit check as 'array length is == 1'~~

edit: 
I miss-understood. What you are referring to is that there is no need to re-order if removing the last element in the array.

I can add a check `if (index == last)` and delete without re-ordering. I don't think it makes a big difference from a gas perspective.

File: contracts/ModuleRegistry.sol (hunk 2 above)
Comment (pabloruiz55) [reply contracts/ModuleRegistry.sol]: I mean, moduleFactoryType or _type, don't want to be introducing different names.

**Comment by pabloruiz55** [comment]
In response to this: 

> Please double check logic for dealing with array index 0. My approach has been to set index 0 of the array to the 0x0 address and prevent from removing, the side effect is that getting the list of module factories will always include the 0x0 address.

I meant instead of adding an item to index 0, change the logic so if there's only one item you don't re-order but just remove that item.

File: contracts/ModuleRegistry.sol (hunk 2 above)
Comment (thegostep) [reply contracts/ModuleRegistry.sol]: okay fixing

**Comment by pabloruiz55** [comment]
Looks good to me now. @adamdossa  please review.

**Review by pabloruiz55 — APPROVED** [review]

**Pull request merged**